File: 백준/Silver/12871. 무한 문자열/무한 문자열.py
s = input()
t = input()

def GCD(a, b):
    while b > 0:
        a, b = b, a%b
    return a

l = len(s)*len(t)//GCD(len(s), len(t))
ss, tt = '', ''
for _ in range(l//len(s)):
    ss += s
for _ in range(l//len(t)):
    tt += t
print(1 if tt==ss else 0)

File: 백준/Silver/12871. 무한 문자열/무한 문자열.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});


function GCD(a, b){
  while (b > 0) {
    [a, b] = [b, a%b];
  }
  return a;
}


rl.question('', (inputS) => {
  const s = inputS;
  rl.question('', (inputT) => {
    const t = inputT;
    const l = s.length*t.length/GCD(s.length, t.length);
    console.log(s.repeat(l/s.length) === t.repeat(l/t.length) ? 1 : 0)
    rl.close();
  })
})